RockPaperScissors/RockPaperScissors/main.swift

```diff
@@ -2,9 +2,94 @@
 //  RockPaperScissors - main.swift
 //  Created by tacocat.
 //  Copyright © tastycode. All rights reserved.
-// 
+//
 
 import Foundation
 
-print("Hello, World!")
+var isExitCondition: Bool = false
+
+enum RockPaperScissorsError: Error
+{
+    case invalidInput
+}
+
+while true
+{
+    print("가위(1), 바위(2), 보(3)! <종료 : 0> : ", terminator: "")
+
+    do
+    {
+        try playRockPaperScissors()
+    }
+    catch RockPaperScissorsError.invalidInput
+    {
+        print("잘못된 입력입니다. 다시 입력해주세요")
+    }
+
+    if isExitCondition
+    {
+        print("----------------------------------------")
+        break
+    }
+}
+
+func playRockPaperScissors() throws
+{
+    let user: Int = try checkInput()
+
+    switch user
+    {
+        case 0:
+            print("게임 종료")
+            isExitCondition.toggle()
+        case 1...3:
+            getGameResult(of: user)
+        default:
+            throw RockPaperScissorsError.invalidInput
+    }
+}
+
+func checkInput() throws -> Int
+{
+    guard let user = readLine(), let user = Int(user)
+    else
```

Comment:
```suggestion
    guard let user = readLine(), let user = Int(user) else {
           throw RockPaperScissorsError.invalidInput
    }
```
위와 같은 형식으로 guard 라인을 바꿔주는 편입니다.